### PR TITLE
Add selectionBehavior option of 'any'

### PR DESCRIPTION
**Pull request by mattvarghese**
Added "any" and "all" as options for selectionBehavior. Also cleaned up to remove redundant language.

### CODE REVIEW

**Comment by alexander-ivanov** [comment]
We were asked for some use-cases for 'all' and 'any' types of selection behavior. We use cds-hooks with own type of cds-hooks, this hook returns several independent actions which can be performed. We can encode them as a permutation of all actions, each permutation as separate suggestions, and use selection behavior - 'at-most-one' but it produces very large output and hurts performance. So we would like to encode each action as a separate suggestion and give the user the possibility to take several of them.

@Sergey778 may want to add some comments.

File: docs/specification/1.0.md [hunk 1]
```diff
@@ -452,7 +452,7 @@ Field | Optionality | Type | Description
...
-`selectionBehavior` | OPTIONAL | *string* | Describes the intended selection behavior of the suggestions in the card. Currently, the only allowed value is `at-most-one`, indicating that the user may choose none or at most one of the suggestions. Future versions of the specification may expand this behavior, so CDS Clients that do not understand the value MUST treat the card as an error. CDS Clients MUST support the value of `at-most-one`.
+`selectionBehavior` | OPTIONAL | *string* | Describes the intended selection behavior of the suggestions in the card. Allowed values are: `at-most-one`, indicating that the user may choose none or at most one of the suggestions, `any`, indicating that the end user may choose any number of suggestions including none of them and all of them, and `all`, indicating that all suggestions MUST be chosen when accepting the card. CDS Clients that do not understand the value MUST treat the card as an error.
```
Comment (jmandel) [inline docs/specification/1.0.md]:
```suggestion
`selectionBehavior` | OPTIONAL | *string* | Describes the intended selection behavior of the suggestions in the card. Allowed values are: `at-most-one`, indicating that the user may choose none or at most one of the suggestions;`any`, indicating that the end user may choose any number of suggestions including none of them and all of them. CDS Clients that do not understand the value MUST treat the card as an error.
```

The desired "all" behavior can be achieved with existing functionality, by listing multiple `actions` in a single suggestion's actions array. @mattvarghese can you review and merge this tweak if you agree?

**Comment by mattvarghese** [comment]
I've updated the PR per Josh's suggested change, to only include 'at-most-one' and 'any', and removed 'all' option.

**Comment by mattvarghese** [comment]
@isaacvetter , @brynrhodes , @dennispatterson - requesting reviews :)

File: docs/specification/1.0.md [hunk 2]
```diff
@@ -452,7 +452,7 @@ Field | Optionality | Type | Description
...
-`selectionBehavior` | OPTIONAL | *string* | Describes the intended selection behavior of the suggestions in the card. Currently, the only allowed value is `at-most-one`, indicating that the user may choose none or at most one of the suggestions. Future versions of the specification may expand this behavior, so CDS Clients that do not understand the value MUST treat the card as an error. CDS Clients MUST support the value of `at-most-one`.
+`selectionBehavior` | OPTIONAL | *string* | Describes the intended selection behavior of the suggestions in the card. Allowed values are: `at-most-one`, indicating that the user may choose none or at most one of the suggestions;`any`, indicating that the end user may choose any number of suggestions including none of them and all of them. CDS Clients that do not understand the value MUST treat the card as an error.
```
Comment (dennispatterson) [inline docs/specification/1.0.md]: Any concerns with removing a 1.0-spec requirement that 'at-most-one' be supported?  I understand it was required as being the only option, but trying to think through any potential issues with removing something that was stated as a requirement.

Comment (mattvarghese) [reply docs/specification/1.0.md]: I'm in fine with the above wording. Do I need to do anything now - I don't see an "accept changes" button or anything.

Comment (isaacvetter) [reply docs/specification/1.0.md]: Hey guys,

`selectionBehavior` is OPTIONAL. Making is REQUIRED would be a breaking change. If no selectionBehavior is provided, how does a client know what behavior to enact?

Isaac

Comment (dennispatterson) [reply docs/specification/1.0.md]: Our call on 11/13 walked through my initial comment and I think we clarified.

Not sure I understand your comment, Isaac, on the optionality of selectionBehavior and "If no selectionBehavior is provided, how does a client know what behavior to enact?"  I know you're familiar with the [conditionally required](https://github.com/cds-hooks/docs/issues/509) aspect of selectionBehavior, so can you clarify?

**Pull request merged**